actionpack/lib/action_dispatch/system_testing/test_helpers/setup_and_teardown.rb

```diff
@@ -17,8 +17,11 @@ def before_setup
         end
 
         def after_teardown
-          take_failed_screenshot
-          Capybara.reset_sessions!
+          begin
+            take_failed_screenshot
+          ensure
+            Capybara.reset_sessions!
```

Comment:
Can't we merge with the `ensure` below?

Reply:
If you mean:
```ruby
def after_teardown
  take_failed_screenshot
ensure
  Capybara.reset_sessions!
  super
end
```

I think we can't, because it needed to resolve bug described in https://github.com/rails/rails/pull/32293